dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java

```diff
@@ -455,7 +458,10 @@ private void taskFinished(TaskInstance taskInstance) {
                     killAllTasks();
                 }
             }
+        } else if (taskInstance.getState().typeIsFinished()) {
```

Comment:
When will this branch be taken? Does it need to do others processing? like task success or fail.

Reply:
It will check in ```org.apache.dolphinscheduler.server.master.runner.WorkflowExecuteThread#getCompleteTaskInstanceMap``` when taskInstance state is cancel(Stop || Kill). This logic exists in 204 branch, but is missing in dev branch.